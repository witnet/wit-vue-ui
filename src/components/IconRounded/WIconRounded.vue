<template>
  <div
    class="border border-black-950 w-min rounded-full p-md bg-wit-blue-500 hover:bg-wit-blue-300 transition-colors"
  >
    <slot classs=""></slot>
  </div>
</template>

<style>
path {
  @apply fill-black-950;
}
</style>
