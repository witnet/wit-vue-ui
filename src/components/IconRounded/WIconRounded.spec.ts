import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import WIconRounded from './WIconRounded.vue'

describe('WIconRounded', () => {
  it('renders properly', () => {
    const wrapper = mount(WIconRounded, {
      slots: {
        default: `
          <WIconRounded v-bind="args">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" id="email">
              <g data-name="Layer 2">
              <path
                  fill="#ccc"
                  d="M19 4H5a3 3 0 0 0-3 3v10a3 3 0 0 0 3 3h14a3 3 0 0 0 3-3V7a3 3 0 0 0-3-3zm0 2-6.5 4.47a1 1 0 0 1-1 0L5 6z"
                  data-name="email"
                ></path>
              </g>
            </svg>
          </WIconRounded>
        `
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders the slot', () => {
    const wrapper = mount(WIconRounded, {
      slots: {
        default: `
          <WIconRounded v-bind="args">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" id="email">
              <g data-name="Layer 2">
              <path
                  fill="#ccc"
                  d="M19 4H5a3 3 0 0 0-3 3v10a3 3 0 0 0 3 3h14a3 3 0 0 0 3-3V7a3 3 0 0 0-3-3zm0 2-6.5 4.47a1 1 0 0 1-1 0L5 6z"
                  data-name="email"
                ></path>
              </g>
            </svg>
          </WIconRounded>
        `
      }
    })

    expect(wrapper.html().replaceAll(' ', '').replaceAll('\n', '')).toContain(
      `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" id="email">
          <g data-name="Layer 2">
          <path
              fill="#ccc"
              d="M19 4H5a3 3 0 0 0-3 3v10a3 3 0 0 0 3 3h14a3 3 0 0 0 3-3V7a3 3 0 0 0-3-3zm0 2-6.5 4.47a1 1 0 0 1-1 0L5 6z"
              data-name="email"
            ></path>
          </g>
        </svg>
        `
        .replaceAll(' ', '')
        .replaceAll('\n', '')
    )
  })
})
