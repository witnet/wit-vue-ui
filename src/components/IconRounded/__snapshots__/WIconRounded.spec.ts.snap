// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WIconRounded > renders properly 1`] = `
<div
  class="border border-black-950 w-min rounded-full p-md bg-wit-blue-500 hover:bg-wit-blue-300 transition-colors"
>
  
  <wiconrounded>
    <svg
      id="email"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        data-name="Layer 2"
      >
        <path
          d="M19 4H5a3 3 0 0 0-3 3v10a3 3 0 0 0 3 3h14a3 3 0 0 0 3-3V7a3 3 0 0 0-3-3zm0 2-6.5 4.47a1 1 0 0 1-1 0L5 6z"
          data-name="email"
          fill="#ccc"
        />
      </g>
    </svg>
  </wiconrounded>
  
</div>
`;
